x-pack/plugins/apm/public/components/app/correlations/use_failed_transactions_correlations_search_strategy.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useRef, useState } from 'react';
+import { Subscription } from 'rxjs';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
+import { ApmPluginStartDeps } from '../../../plugin';
+import {
+  IKibanaSearchRequest,
+  IKibanaSearchResponse,
+  isCompleteResponse,
+  isErrorResponse,
+} from '../../../../../../../src/plugins/data/common';
+import { BaseSearchStrategyResponse } from '../../../../common/search_strategies/failure_correlations/types';
+
+interface SearchStrategyOptions {
+  environment?: string;
+  kuery?: string;
+  serviceName?: string;
+  transactionName?: string;
+  transactionType?: string;
+  start?: string;
+  end?: string;
+}
+
+export function useFailedTransactionsCorrelationsSearchStrategy<
+  T extends BaseSearchStrategyResponse
+>(params: SearchStrategyOptions, strategy: string) {
+  const {
+    services: { data },
+  } = useKibana<ApmPluginStartDeps>();
+
+  const [error, setError] = useState<Error>();
```

Comment:
For the similar code for latency correlations we decided to reduce the number of required state handlers and maintain a single state object, you can see how it was done here: https://github.com/elastic/kibana/pull/107266/commits/128468767afd048cbb5c7b5a1ef4b547febb7828

Reply:
@walterra Thanks for the heads up - I've also refactored to use the state handler and t be more inline with Latency correlation changes [here](https://github.com/elastic/kibana/pull/108441/commits/6bbb68ca392c4d745fe2acb696f21413c09c6b56)